### PR TITLE
Fixes #23191 - replace dynamic controller routing

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Review by ohadlevy — COMMENTED** [review]
:+1: overall, I do wonder if perhaps it makes more sense to use a helper, and keep the welcome url as part of the current routing specific block, e.g. something like

```ruby
resources :architectures
welcome :architectures
```

This way, when we edit/remove it, its all in one place?

I will not block of course based on this comment :)

File: config/routes.rb [hunk 1]
```diff
@@ -14,7 +14,42 @@
     end
   end
 
-  get '(:controller)/help', :action => 'welcome', :as => "help"
+  welcoming_controllers = [
```
Comment (timogoebel) [inline config/routes.rb:17]: Do plugins need to extend this list?

Comment (ohadlevy) [reply config/routes.rb:17]: oh yes, https://github.com/search?q=org%3Atheforeman+welcome.html.erb&type=Code

**Comment by mmoll** [comment]
test failures related :man_shrugging:

**Comment by mmoll** [comment]
All :green_heart: now. I admit this was just the by-product of nightly hackery, but I still think it would be OK to go in as-is. If anybody has a better way to handle everything in one place (also including the permissions), please open a PR against this branch in my fork, as this is not hightly time-critical to arrive in develop.

**Comment by mmoll** [comment]
As one month passed by without further enhancements, I'm kindly nudging to get this in as-is.

File: config/routes.rb (hunk 1 above)
Comment (ohadlevy) [reply config/routes.rb:17]: @mmoll I think this issue still stands, or at least the way I understand it users of a plugin with a welcome page will get 403 ? happy to be wrong..

Comment (mmoll) [reply config/routes.rb:17]: @ohadlevy correct, the affected plugins will need updates after this gets in.

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**